middleman-core/lib/middleman-core/preview_server/information.rb

```diff
@@ -0,0 +1,155 @@
+require 'active_support/core_ext/object/blank'
```

Comment:
This one wraps how the information is retrieved. This gaves me the most headaches. Each class is for a specific use case. The validation checks are still missing. I will add them later/tomorrow.